src/server/system_services/bucket_server.js

```diff
@@ -151,6 +153,36 @@ async function create_bucket(req) {
             );
             changes.insert.tieringpolicies = [tiering_policy];
             changes.insert.tiers = [tier];
+
+            // Attach a `GLACIER` tier to the bucket if it is not namespace.caching
+            // and appropriate configuration is set
+            if (!req.rpc_params.namespace?.namespace?.caching) {
```

Comment:
Isn't it a double namespace there?
But also, this also needs to prevent applying this to any namespace buckets (not just ones with caching option set).

```suggestion
            if (!req.rpc_params.namespace) {
```